enhancing/enhancer.js

```diff
@@ -2,21 +2,31 @@ module.exports = {
   succeed,
   fail,
   repair,
-  get,
-};
+  get
+}
 
 function succeed(item) {
-  return { ...item };
+  if (item.enchantment <= 19) {
+    item.enchantment += 1
+  }
+  return { ...item }
 }
 
 function fail(item) {
-  return { ...item };
+  if (item.enchantment > 16) {
+    item.durability -= 10
+    item.enchantment -= 1
+  } else if (item.enchantment > 15) item.durability -= 10
+  else item.durability -= 5
+  if (item.durability < 0) item.durability = 0
```

Comment:
This all is directly modifying the item, but then you return a copy of the results. Is this intended? In order to make this function pure, you could use object destructuring to make it convenient:
```javascript
function fail({ enchantment, durability }) {
  if (enchantment > 16) {
    durability -= 10
    enchantment -= 1
  } else if (enchantment > 15) durability -= 10
  else durability -= 5
  if (durability < 0) durability = 0
  return { enchantment, durability }
}
```

Reply:
The same applies to other functions where you directly alter the passed in item object!